src/ui/public/timefilter/timefilter.js

```diff
@@ -20,142 +20,157 @@
+  return obj.isValid() ? obj : stringTime;
+}
+
+const registerWithGlobalState = _.once((globalState) => {
+  const uiSettings = chrome.getUiSettingsClient();
+  const timeDefaults = uiSettings.get('timepicker:timeDefaults');
+  const refreshIntervalDefaults = uiSettings.get('timepicker:refreshIntervalDefaults');
+
+  timefilter.setTime(_.defaults(globalState.time || {}, timeDefaults));
+  timefilter.setRefreshInterval(_.defaults(globalState.refreshInterval || {}, refreshIntervalDefaults));
+
+  globalState.on('fetch_with_changes', () => {
+  // clone and default to {} in one
+    const newTime = _.defaults({}, globalState.time, timeDefaults);
+    const newRefreshInterval = _.defaults({}, globalState.refreshInterval, refreshIntervalDefaults);
+
+    if (newTime) {
+      if (newTime.to) newTime.to = convertISO8601(newTime.to);
+      if (newTime.from) newTime.from = convertISO8601(newTime.from);
     }
 
-    Timefilter.prototype.update = function () {
-      $rootScope.$apply();
-    };
-
-    Timefilter.prototype.getForceNow = function () {
-      const query = $location.search().forceNow;
-      if (!query) {
-        return;
-      }
-
-      const ticks = Date.parse(query);
-      if (isNaN(ticks)) {
-        throw new Error(`forceNow query parameter can't be parsed`);
-      }
-      return new Date(ticks);
-    };
-
-    Timefilter.prototype.get = function (indexPattern, timeRange) {
-      return getTime(indexPattern, timeRange ? timeRange : this.time, this.getForceNow());
-    };
-
-    Timefilter.prototype.calculateBounds = function (timeRange) {
-      return calculateBounds(timeRange, { forceNow: this.getForceNow() });
-    };
-
-    Timefilter.prototype.getBounds = function () {
-      return this.calculateBounds(this.time);
-    };
-
-    Timefilter.prototype.getActiveBounds = function () {
-      if (this.isTimeRangeSelectorEnabled) {
-        return this.getBounds();
-      }
-    };
-
-    /**
-     * Show the time bounds selector part of the time filter
-     */
-    Timefilter.prototype.enableTimeRangeSelector = function () {
-      this.isTimeRangeSelectorEnabled = true;
-    };
-
-    /**
-     * Hide the time bounds selector part of the time filter
-     */
-    Timefilter.prototype.disableTimeRangeSelector = function () {
-      this.isTimeRangeSelectorEnabled = false;
-    };
-
-    /**
-     * Show the auto refresh part of the time filter
-     */
-    Timefilter.prototype.enableAutoRefreshSelector = function () {
-      this.isAutoRefreshSelectorEnabled = true;
-    };
-
-    /**
-     * Hide the auto refresh part of the time filter
-     */
-    Timefilter.prototype.disableAutoRefreshSelector = function () {
-      this.isAutoRefreshSelectorEnabled = false;
-    };
-
-    return new Timefilter();
+    timefilter.setTime(newTime);
+    timefilter.setRefreshInterval(newRefreshInterval);
+  });
+});
+
+uiRoutes
+  .addSetupWork((globalState) => {
+    return registerWithGlobalState(globalState);
```

Comment:
I'm just realizing that since we are binding the timefilter to global state we haven't totally disconnected it from angular... maybe it makes sense then for the timefilter to ensure it is in a digest cycle before it does any mutations, though that's not really a pattern I would like to be used elsewhere...